docs/sources/mimir/operators-guide/deploy-grafana-mimir/jsonnet/migrate-from-microservices-to-read-write.md

```diff
@@ -0,0 +1,311 @@
+kubectl get -o yaml statefulset store-gateway-zone-b > store-gateway-zone-b.yaml; yq eval -i '.spec' store-gateway-zone-b.yaml
+kubectl get -o yaml statefulset store-gateway-zone-c > store-gateway-zone-c.yaml; yq eval -i '.spec' store-gateway-zone-c.yaml
+kubectl get -o yaml statefulset mimir-write-zone-a > mimir-write-zone-a.yaml; yq eval -i '.spec' mimir-write-zone-a.yaml
+kubectl get -o yaml statefulset mimir-write-zone-b > mimir-write-zone-b.yaml; yq eval -i '.spec' mimir-write-zone-b.yaml
+kubectl get -o yaml statefulset mimir-write-zone-c > mimir-write-zone-c.yaml; yq eval -i '.spec' mimir-write-zone-c.yaml
+kubectl get -o yaml statefulset mimir-backend-zone-a > mimir-backend-zone-a.yaml; yq eval -i '.spec' mimir-backend-zone-a.yaml
+kubectl get -o yaml statefulset mimir-backend-zone-b > mimir-backend-zone-b.yaml; yq eval -i '.spec' mimir-backend-zone-b.yaml
+kubectl get -o yaml statefulset mimir-backend-zone-c > mimir-backend-zone-c.yaml; yq eval -i '.spec' mimir-backend-zone-c.yaml
+
+# Diff deployments and statefulsets:
+
+## Write
+diff --color=always distributor.yaml mimir-write-zone-a.yaml
+diff --color=always ingester-zone-a.yaml mimir-write-zone-a.yaml
+diff --color=always ingester-zone-b.yaml mimir-write-zone-b.yaml
+diff --color=always ingester-zone-c.yaml mimir-write-zone-c.yaml
+
+## Read
+diff --color=always query-frontend.yaml mimir-read.yaml
+diff --color=always querier.yaml mimir-read.yaml
+diff --color=always ruler-query-frontend.yaml mimir-read.yaml
+diff --color=always ruler-querier.yaml mimir-read.yaml
+
+## Backend
+diff --color=always overrides-exporter.yaml mimir-backend-zone-a.yaml
+diff --color=always query-scheduler.yaml mimir-backend-zone-a.yaml
+diff --color=always ruler-query-scheduler.yaml mimir-backend-zone-a.yaml
+diff --color=always ruler.yaml mimir-backend-zone-a.yaml
+diff --color=always compactor.yaml mimir-backend-zone-a.yaml
+diff --color=always store-gateway-zone-a.yaml mimir-backend-zone-a.yaml
+diff --color=always store-gateway-zone-b.yaml mimir-backend-zone-b.yaml
+diff --color=always store-gateway-zone-c.yaml mimir-backend-zone-c.yaml
+
+```
+
+## Step 3: Migrate read-path to read-write service
+
+Scale up the Mimir read component using either the autoscaler or explicitly setting the number of replicas. (Keep the current microservices at their present level of replicas or autoscaling).
+
+```jsonnet
+{
+  _config+:: {
+    deployment_mode: 'migration',
+
+    mimir_write_replicas: 0,
+    mimir_read_replicas: 3,
+    mimir_backend_replicas: 0,
+    autoscaling_mimir_read_enabled: false,
+  },
+}
+```
+
+```jsonnet
+{
+  _config+:: {
+    deployment_mode: 'migration',
+
+    mimir_write_replicas: 0,
+    mimir_backend_replicas: 0,
+    autoscaling_mimir_read_enabled: true,
+    autoscaling_mimir_read_min_replicas: 3,
+    autoscaling_mimir_read_max_replicas: 30,
+  },
+}
+```
+
+The Read-write querier at this point will start running queries from the `query-scheduler` (as they share the same ring).
```

Comment:
I would organize these sub-steps is in a numbered list. They will be easier to consume.